### PR TITLE
Add partial scoring support

### PR DESCRIPTION
**Pull request by benburton**

### CODE REVIEW

File: controller/src/scoring.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+function defaultScore(question, session) {
+  let flattenedCorrect = question.correctResponse.find((response) => response instanceof Object) === undefined ? question.correctResponse :
+    question.correctResponse.map(({id}) => id);
+  let allCorrect = _.isEqual(_.cloneDeep(session.value).sort(), _.cloneDeep(flattenedCorrect).sort());
+  return allCorrect ? 1 : 0;
+}
+ 
+function partialScore(question, session) {
+  let flattenedCorrect = question.correctResponse.find((response) => response instanceof Object) === undefined ? question.correctResponse :
```
Comment (edeustace) [inline controller/src/scoring.js]: can we make this a shared function it's used here and in src/index.js

**Review by edeustace — COMMENTED** [review]
:+1:

File: controller/src/scoring.js (hunk 1 above)
Comment (benburton) [reply controller/src/scoring.js]: Yeah, good idea.

**Comment by edeustace** [comment]
:+1:

**Pull request merged**